packages/design/src/DataTable/Cells.tsx

```diff
@@ -2,26 +2,60 @@ import React from 'react';
 import { Label } from 'design';
 import * as Icons from 'design/Icon';
 import { displayDate } from 'shared/services/loc';
-import { SortDir } from './types';
+import { ServersideConfig, SortDir, TableColumn } from './types';
 
 export const Cell = props => <td children={props.children} {...props} />;
 
-export const SortHeaderCell = ({ dir, text, onClick }: SortHeaderCellProps) => {
+export function SortHeaderCell<T>({
+  column,
+  serverside,
+  dir,
+  text,
+  onClick,
+}: SortHeaderCellProps<T>) {
+  function handleClick() {
+    if (serverside?.sort) {
+      serverside.setSort({
+        dir: serverside.sort.dir === 'ASC' ? 'DESC' : 'ASC',
+        fieldName: column.key,
+      });
+    } else {
+      onClick();
```

Comment:
What's going on here? This component either accepts a custom `onClick` or does sorting, but not both?

Reply:
Yes, the custom `onClick` is used when we want client-side sorting, but if the Table is `serverside` then we want the sorting to be handled by the back. I will refactor this to try to make it a little more clear.